docs/service-guide/update-pull-secret.rst

```diff
@@ -0,0 +1,43 @@
+######################################################
+Updating the pull secret stored in 1Password and Vault
+######################################################
+
+The pull secret, present in each RSP instance, and shared by many
+services there, is notoriously tricky to format correctly.
+
+The recommended way to update it is to edit the pull secret in 1Password
+and then deploy it with the `installer/update-secrets.sh` script;
+however, this only works (at the time of writing, 20 May 2022) on Linux
+systems with the 1Password 1.x CLI installed.
+
+If you need to update the pull secret manually for an environment, here
+are the important things to know:
+
+You will first set the necessary environment variables:
+
+* ``VAULT_ADDR`` must be set to ``https://vault.lsst.codes``
+* ``VAULT_TOKEN`` must be set to the appropriate write token for the RSP
+  instance.
+
+Then you will construct the updated secret in a file; for purposes of
+this example, let's call it ``pull-secret.json``.  It should look like
+this::
+
+  { ".dockerconfigjson": "{\"auths\": {\"ghcr.io\": {\"auth\": \"base64string\", \"password\": \"cleartexttoken\",\"username\": \"token\"},\"index.docker.io\": {\"auth\":\"base64string\",\"password\":\"cleartextpassword\",\"username\":\"sqrereadonly\"}}}}
+
+In short: the value is a *string* (not a JSON object) with all keys and
+values quoted with backslash-escaped double quotes.
+
+Once you have this file created, run:
+
+``vault kv put secret/k8s_operator/<environment>/pull-secret @pull-secret.json``
```

Comment:
I would not modify secrets this way, since it requires this extra level of encoding and JSON. The way I always update secrets in Vault (not just this secret, every secret), is:

#. Highlight the correct secret value and copy it to my paste buffer.
#. Start with `vault kv patch secret/k8s_operator/<environment>/pull-secret .dockerconfigjson='` (note the single quote)
#. Paste the secret into the command line
#. Type `'` and press Enter

I feel like most of the problems we've had with Vault secret management is using ways of updating or viewing secrets that introduce another level of JSON, either using JSON output mode rather than the default or using this syntax to modify the Vault secret that takes JSON input. I would just avoid all use of JSON with Vault; it seems to cause problems.

Reply:
I've updated the doc with your technique, which I agree is better than mine.